### PR TITLE
Add runtime selection as parameter to Makefile

### PR DESCRIPTION
**Pull request by MaxBab**
Runtime selection bacome a parameter.
By default, it will use "docker".
A user could change the runtime engine to be used with:

make env-deploy CONF=... RUNTIME=docker
make env-deploy CONF=... RUNTIME=podman

### CODE REVIEW

File: scripts/local_environment.sh [hunk 1]
```diff
@@ -53,22 +65,22 @@ function destroy_container() {
...
                 shift
                 ;;
-            --get-runtime)
-                check_container_engine
-                shift
+            --runtime)
+                if [[ -n "$2" ]]; then
+                    RUNTIME="$2"
+                    shift 2
+                fi
```
Comment (skitt) [inline scripts/local_environment.sh:83]: This needs to fail if there is no `$2`, otherwise the script will get stuck in a loop ;-).

**Review by skitt — CHANGES_REQUESTED** [review]

File: scripts/local_environment.sh [hunk 2]
```diff
@@ -83,5 +95,26 @@ function main() {
     done
 }
 
+function main() {
+    RUN_COMMAND=deploy
+    parse_arguments "$@"
```
Comment (skitt) [inline scripts/local_environment.sh:100]: The default `RUNTIME` should be set here too

File: scripts/local_environment.sh [hunk 3]
```diff
@@ -18,21 +18,33 @@ function usage() {
...
-    elif command -v docker &> /dev/null; then
-        RUNTIME=docker
+function sets_container_engine() {
+    if [[ "$RUNTIME" == "docker" ]]; then
+        if [[ -x "$(command -v docker)" && -S /var/run/docker.sock ]]; then
+            RUNTIME="docker"
+        else
+            echo "Docker engine is not available"
+            exit 1
+        fi
```
Comment (skitt) [inline scripts/local_environment.sh:35]: `RUNTIME` is already set, so the `if` branch isn’t needed

```suggestion
        if ! [[ -x "$(command -v docker)" && -S /var/run/docker.sock ]]; then
            echo "Docker engine is not available"
            exit 1
        fi
```

This applies to the `podman` test too.

File: scripts/local_environment.sh [hunk 4]
```diff
@@ -18,21 +18,33 @@ function usage() {
...
     --help|-h       - Print help
 EOF
 }
 
-function check_container_engine() {
-    if command -v podman &> /dev/null; then
-	RUNTIME=podman
-    elif command -v docker &> /dev/null; then
-        RUNTIME=docker
+function sets_container_engine() {
```
Comment (skitt) [inline scripts/local_environment.sh:28]: This only checks whether the chosen container engine is actually available, so

```suggestion
function check_container_engine() {
```

still seems appropriate.

File: scripts/local_environment.sh (hunk 1 above)
Comment (MaxBab) [reply scripts/local_environment.sh:83]: Added failure in case no value provided.

File: scripts/local_environment.sh (hunk 2 above)
Comment (MaxBab) [reply scripts/local_environment.sh:100]: The RUNTIME already defaulted at line number 3 alongside with other default vars.

File: scripts/local_environment.sh (hunk 3 above)
Comment (MaxBab) [reply scripts/local_environment.sh:35]: Fixed.

File: scripts/local_environment.sh (hunk 4 above)
Comment (MaxBab) [reply scripts/local_environment.sh:28]: ack.

File: scripts/local_environment.sh (hunk 2 above)
Comment (skitt) [reply scripts/local_environment.sh:99]: Ah yes, sorry, I had only checked the diff.

**Review by skitt — APPROVED** [review]

**Pull request merged**